deployer/src/main/java/eu/seaclouds/policy/SeaCloudsInitializerPolicy.java

```diff
@@ -0,0 +1,544 @@
+
+    }
+
+    private Effector<Void> newStartEffector() {
+        return Effectors.effector(Startable.START)
+                .impl(new EffectorBody<Void>() {
+                    @Override
+                    public Void call(ConfigBag parameters) {
+                        LOG.info("Starting SeaCloudsInitializerPolicy " + "for " + entity.getId());
+                        installSlaAgreement();
+                        installMonitoringRules();
+                        notifyRulesReady();
+                        installInfluxDbObservers();
+                        installGrafanaDashboards();
+
+                        // Rewire the original behaviour
+                        Collection<? extends Location> locations = null;
+                        Object locationRaw = parameters.getStringKey(EffectorStartableImpl.StartParameters.LOCATIONS.getName());
+                        locations = Locations.coerceToCollection(getManagementContext(), locationRaw);
+                        ((StartableApplication) entity).start(locations);
+
+                        return null;
+                    }
+                })
+                .build();
+    }
+
+    private Effector<Void> newStopEffector() {
+        return Effectors.effector(Startable.STOP)
+                .parameter(SoftwareProcess.StopSoftwareParameters.STOP_PROCESS_MODE)
+                .parameter(SoftwareProcess.StopSoftwareParameters.STOP_MACHINE_MODE)
+                .impl(new EffectorBody<Void>() {
+                    @Override
+                    public Void call(ConfigBag parameters) {
+                        LOG.info("Stopping SeaCloudsInitializerPolicy " + "for " + entity.getId());
+                        removeSlaAgreement();
+                        removeMonitoringRules();
+                        removeInfluxDbObservers();
+                        removeGrafanaDashboard();
+
+                        // Rewire the original behaviour
+                        ((StartableApplication) entity).stop();
+
+                        return null;
+                    }
+                })
+                .build();
+    }
+
+    private void installSlaAgreement() {
+        LOG.info("SeaCloudsInitializerPolicy is installing SLA Agreements for " + entity.getId());
+
+        HttpToolResponse response = post(getConfig(SLA_ENDPOINT) + "/seaclouds/agreements", MediaType.APPLICATION_XML_UTF_8.toString(),
+                getConfig(SLA_USERNAME), getConfig(SLA_PASSWORD), BaseEncoding.base64().decode(getConfig(SLA_AGREEMENT)));
+
+        if (!HttpTool.isStatusCodeHealthy(response.getResponseCode())) {
+            throw new RuntimeException("Something went wrong during the SLA Agreements installation. " +
+                    "Invalid response code, " + response.getResponseCode() + ":" + response.getContentAsString());
+        } else {
+            entity.sensors().set(SLA_ID, agreement.getAgreementId());
```

Comment:
why do you want to store this `agreeementId` in the application?
also the method is called `installSLA` but this does more, no?